lib/Backend/CodeGenNumberAllocator.cpp

```diff
@@ -324,6 +324,9 @@ Js::JavascriptNumber* XProcNumberPageSegmentImpl::AllocateNumber(Func* func, dou
                 Recycler::FillPadNoCheck(pLocalNumber, sizeof(Js::JavascriptNumber), sizeCat, false);
                 pLocalNumber = new (pLocalNumber) Js::JavascriptNumber(localNumber);
             }
+#else
+            Assert(sizeCat == sizeof(Js::JavascriptNumber));
+            __analysis_assume(sizeCat == sizeof(Js::JavascriptNumber));
```

Comment:
RECYCLER_MEMORY_VERIFY means it's checked build, which looks we don't run pre-fast with